### PR TITLE
fix Issue 16188 - [REG2.069] ICE on invalid code

### PR DESCRIPTION
**Pull request by WalterBright**
Unsure if this is the right fix. Let's see what the autotester says.

### CODE REVIEW

**Comment by WalterBright** [comment]
The code in AliasDeclaration.semantic() is the most unholy mess I've ever seen.

**Comment by WalterBright** [comment]
I'm not real happy with this fix, as the wrong error message is generated, as described in the attached test case. The whole `AliasDeclaration.semantic()` needs to be redone to make it fully lazy. But this fix at least takes care of the seg fault regression.

File: src/declaration.d [hunk 1]
```diff
@@ -653,19 +653,22 @@ extern (C++) final class AliasDeclaration : Declaration
...
-        inuse = 0;
-        semanticRun = PASSsemanticdone;
-
-        if (auto sx = overnext)
+        else
         {
-            overnext = null;
-            if (!overloadInsert(sx))
-                ScopeDsymbol.multiplyDefined(Loc(), sx, this);
+            semanticRun = PASSsemanticdone;
```
Comment (MartinNowak) [inline src/declaration.d]: Moving `semanticRun` here means repeated semantic for the error case. Is this intended?

File: src/mtype.d [hunk 2]
```diff
@@ -7886,10 +7886,18 @@ extern (C++) final class TypeTypeof : TypeQualified
...
+        exp2 = resolvePropertiesOnly(sc2, exp2);
         sc2.pop();
 
+        if (exp2.op == TOKerror)
+        {
+            if (!global.gag)
+                exp = exp2;
+            goto Lerr;
+        }
+        exp = exp2;
```
Comment (MartinNowak) [inline src/mtype.d]: This fixes the ICE?

**Comment by MartinNowak** [comment]
The overcomplicated alias stuff implementations are also the reason why I got stuck with https://github.com/dlang/dmd/pull/5847.

**Comment by MartinNowak** [comment]
d_do_test doesn't check the return code and silently let's a segfault in fail_compilation pass.

File: src/declaration.d (hunk 1 above)
Comment (MartinNowak) [inline src/declaration.d]: The change of src/declaration.d seems unrelated to the segfault fix, can we remove it?
If you intended to rerun semantic to produce the right error message, that apparently didn't work anyhow.

**Comment by MartinNowak** [comment]
See #5927
